frontend/coprs_frontend/coprs/views/apiv3_ns/schema.py

```diff
@@ -0,0 +1,560 @@
+def field2arg(name, field, **kwargs):
+    """
+    Take a field on the input and create an `Argument` for `RequestParser`
+    based on it.
+    """
+    return Argument(
+        name,
+        type=to_arg_type(field),
+        help=field.description,
+        **kwargs,
+    )
+
+
+def merge_parsers(a, b):
+    """
+    Take two `RequestParser` instances and create a new one, combining all of
+    their arguments.
+    """
+    parser = RequestParser()
+    for arg in a.args + b.args:
+        parser.add_argument(arg)
+    return parser
+
+
+def get_package_parser():
+    # pylint: disable=missing-function-docstring
+    parser = RequestParser()
+    parser.add_argument(field2arg("ownername", ownername_field, required=True))
+    parser.add_argument(field2arg("projectname", projectname_field, required=True))
+    parser.add_argument(field2arg("packagename", packagename_field, required=True))
+
+    parser.add_argument(
+        "with_latest_build", type=boolean, required=False, default=False,
+        help=(
+            "The result will contain 'builds' dictionary with the latest "
+            "submitted build of this particular package within the project"))
+
+    parser.add_argument(
+        "with_latest_succeeded_build", type=boolean, required=False, default=False,
+        help=(
+            "The result will contain 'builds' dictionary with the latest "
+            "successful build of this particular package within the project."))
+
+    return parser
+
+
+def add_package_parser():
+    # pylint: disable=missing-function-docstring
+    args = [
+        # SCM
+        field2arg("clone_url", clone_url_field),
+        field2arg("committish", committish_field),
+        field2arg("subdirectory", subdirectory_field),
+        field2arg("spec", spec_field),
+        field2arg("scm_type", scm_type_field),
+
+        # Rubygems
+        field2arg("gem_name", gem_name_field),
+
+        # PyPI
+        field2arg("pypi_package_name", pypi_package_name_field),
+        field2arg("pypi_package_version", pypi_package_version_field),
+        field2arg("spec_generator", pypi_spec_generator_field),
+        field2arg("spec_template", pypi_spec_template_field),
+        field2arg("python_versions", pypi_versions_field),
+
+        # Custom
+        field2arg("script", custom_script_field),
+        field2arg("builddeps", custom_builddeps_field),
+        field2arg("resultdir", custom_resultdir_field),
+        field2arg("chroot", custom_chroot_field),
+
+
+        field2arg("packagename", packagename_field),
+        field2arg("source_build_method", source_build_method_field),
+        field2arg("max_builds", max_builds_field),
+        field2arg("webhook_rebuild", auto_rebuild_field),
+    ]
+    parser = RequestParser()
+    for arg in args:
+        arg.location = "json"
+        parser.add_argument(arg)
+    return parser
+
+
+def edit_package_parser():
+    # pylint: disable=missing-function-docstring
+    parser = add_package_parser().copy()
```

Comment:
Is this a deep copy?

Reply:
I would say, yes.

> `parser_copy.args = deepcopy(self.args)`

https://flask-restx.readthedocs.io/en/latest/_modules/flask_restx/reqparse.html#RequestParser.copy